lib/Dialect/TritonGPU/Transforms/Pipeliner/TMAStoresPipeline.cpp

```diff
@@ -0,0 +1,94 @@
+#include "Schedule.h"
+#include "triton/Dialect/TritonGPU/IR/Dialect.h"
+#include "triton/Dialect/TritonNvidiaGPU/IR/Dialect.h"
+
+using namespace mlir;
+namespace tt = mlir::triton;
+namespace ttg = mlir::triton::gpu;
+namespace ttng = mlir::triton::nvidia_gpu;
+
+static SmallVector<tt::ExperimentalDescriptorStoreOp>
+getTMAStores(scf::ForOp forOp) {
+  SmallVector<tt::ExperimentalDescriptorStoreOp> tmaStores;
+
+  // Do not use walk, as we don't want to walk into nested loops.
+  std::function<void(Operation *)> collectTMAStores = [&](Operation *op) {
+    if (auto storeOp = dyn_cast<tt::ExperimentalDescriptorStoreOp>(op)) {
+      tmaStores.push_back(storeOp);
+    }
+    for (Region &region : op->getRegions()) {
+      for (Block &block : region) {
+        for (Operation &op : block) {
```

Comment:
nit: you can do `for (Operation &op : region.getOps())`

Reply:
neat!